cpp/src/arrow/compute/kernels/vector_nested.cc

```diff
@@ -76,6 +76,24 @@ struct ListParentIndicesArray {
 
   Status Visit(const LargeListType& type) { return VisitList(type); }
 
+  Status Visit(const FixedSizeListType& type) {
+    const int32_t slot_length = type.list_size();
+    const int64_t values_length = slot_length * (input->length - input->GetNullCount());
+    ARROW_ASSIGN_OR_RAISE(auto indices, ctx->Allocate(values_length * sizeof(int32_t)));
+    int32_t* out_indices = reinterpret_cast<int32_t*>(indices->mutable_data());
+    const uint8_t* bitmap = input->GetValues<uint8_t>(0, 0);
```

Comment:
Probably here, use `auto` and `offset_type`.

Reply:
I'd rather not use too much auto since it gets confusing trying to figure out types of things. I'll use offset_type at least.